apollo-coroutines-support/src/main/kotlin/com/apollographql/apollo/coroutines/CoroutinesExtensions.kt

```diff
@@ -72,13 +76,49 @@ fun <T> ApolloQueryWatcher<T>.toFlow(): Flow<Response<T>> = callbackFlow {
 }
 
 
+/**
+ * Suspends the [ApolloCall] until it completes and returns the value on success or throws an exception on failure.
+ * The [ApolloCall] is cancelled when the coroutine running the operation is cancelled as well.
+ *
+ * This is a convenience method that will only return the first value emitted. If more than one
+ * response is required, for an example to retrieve cached and network response, use [toFlow] instead.
+ *
+ * @param <T>  the value type.
+ * @return the response on success.
+ * @throws ApolloException on failure.
+ */
+suspend fun <T> ApolloCall<T>.await(): Response<T> = suspendCancellableCoroutine { cont ->
+
+  cont.invokeOnCancellation {
+    cancel()
+  }
+
+  enqueue(object : ApolloCall.Callback<T>() {
+
+    private val wasCalled = AtomicBoolean(false)
```

Comment:
is there any reason to not use `cont.isCancelled`? There was an issue with RC but looks like it was fixed https://github.com/Kotlin/kotlinx.coroutines/issues/830

Reply:
This boolean was introduced to guard against calling `resume` multiple times to prevent an exception getting thrown for resuming with multiple values instead of one. Same below as. So I'm not sure how `isCancelled ` would solve this particular case.

Reply:
Sorry, my bad. I misinterpreted the description of the issue.